projects/packages/stats/src/class-wpcom-stats.php

```diff
@@ -399,6 +404,44 @@ protected function fetch_stats( $args = array() ) {
 		return $wpcom_stats;
 	}
 
+	/**
+	 * Fetches stats data from WPCOM or local Cache. Caches locally for 5 minutes.
+	 *
+	 * Unlike the above function, this caches data in the post meta table. As such,
+	 * it prevents wp_options from blowing up when retrieving views for large numbers
+	 * of posts at the same time. However, the final response is the same as above.
+	 *
+	 * @param array $args Query parameters.
+	 * @param int   $post_id Post ID to acquire stats for.
+	 *
+	 * @return array|WP_Error
+	 */
+	protected function fetch_post_stats( $args, $post_id ) {
+		$endpoint    = $this->build_endpoint();
+		$meta_name   = '_' . self::STATS_CACHE_TRANSIENT_PREFIX;
+		$stats_cache = get_post_meta( $post_id, $meta_name );
+
+		if ( $stats_cache ) {
+			$data = reset( $stats_cache );
+
+			if ( is_wp_error( $data ) ) {
+				return $data;
+			}
+
+			$time       = key( $data );
+			$expiration = self::STATS_CACHE_EXPIRATION_IN_MINUTES * MINUTE_IN_SECONDS;
+
+			if ( ( time() - $time ) < $expiration ) {
+				return array_merge( array( 'cached_at' => $time ), $data[ $time ] );
+			}
+		}
```

Comment:
Maybe we could try something like:

```PHP
		if ( is_array( $data ) ) {
			  $time       = key( $data );
			  $expiration = self::STATS_CACHE_EXPIRATION_IN_MINUTES * MINUTE_IN_SECONDS;
  
			  if ( ( time() - $time ) < $expiration ) {
				  return array_merge( array( 'cached_at' => $time ), $data[ $time ] );
			  }
			}
		}
```

Reply:
I tested this a bit more and it seems to work well, so I've committed it now. That way we only cache the view count, which is the data we want to cache.

Happy to change it if you prefer a different approach though!